### ISSUE TITLE
Change mod WTFPL licenses to allow Minetest use in schools etc.

### ISSUE THREAD

**Issue by paramat** [minetest/minetest_game#1259]
See https://forum.minetest.net/viewtopic.php?f=3&t=14246
Also see https://forum.minetest.net/viewtopic.php?f=47&t=15354
WTFPL is missing protections and is dangerous.

Celeron55 recommends CC0 https://forum.minetest.net/viewtopic.php?p=211929#p211929
CC0 is a similar license but with extra legal protections.
https://creativecommons.org/publicdomain/zero/1.0/

With a similar license, and since WTFPL gives a lot of freedom i consider this change reasonable. WTFPL allows relicensing.
But what if we can't contact the mod author? perhaps it's still reasonable to change without consent, since it's WTFPL? I think so, especially since a change adds protection for an author.
What other licenses could be used? MIT?

MIT https://opensource.org/licenses/MIT

Once we make some decisions i'm willing to make the PR itself.

**Comment by rubenwardy** [comment]
Wtfpl also doesn't have a no warranty disclaimer. I'm not sure if cc0 does

**Comment by paramat** [comment]
MIT looks better, a more complete disclaimer.

**Comment by paramat** [comment]
Mods missing licenses:
give initial stuff
sethome
walls

Most are WTFPL.

**Comment by Calinou** [comment]
> Wtfpl also doesn't have a no warranty disclaimer. I'm not sure if cc0 does

CC0 does have a warranty disclaimer. However, it also explictly denies patent rights (which is why the OSI has not approved it).

**Comment by paramat** [comment]
https://opensource.org/licenses/category

**Comment by sofar** [comment]
@paramat If I was unclear in any of my contributions, I prefer that the code is licensed LGPL-2.1 as much as possible. If we settle on MIT, I'd be very OK with that, and I certainly would allow relicensing to both those without any discussion.

**Comment by paramat** [comment]
You can have LGPL for your walls mod no problem.

**Comment by paramat** [comment]
Tenplus1 wants a free license for  'sethome' mod http://minetest.org/ircproject/2016-08-19.html

**Comment by C1ffisme** [comment]
If anything, we should remove all profanity, etc. from Minetest itself. Anyone can tell you that 3rd party mods will sometimes have some weird stuff going on, but at least the main game should be safe for all.

Plus we avoid someone suing us because some image didn't have a warranty, and they somehow used it to injure their property.

**Comment by paramat** [comment]
Discussion http://irc.minetest.ru/minetest-dev/2016-08-19#i_4683676
MIT seems the favourite.

**Comment by pithydon** [comment]
I thought all contributions to Minetest Game were required to be relicenseable to LGPL.

**Comment by rubenwardy** [comment]
Is MIT relicensable to LGPL?

**Comment by Calinou** [comment]
> Is MIT relicensable to LGPL?

Yes, MIT is (L)GPL-compatible since it allows for sublicensing.

**Comment by paramat** [comment]
If you are tagged here please let us know what license you would like for your mod, either MIT or LGPL.
Current WTFPL mods:

@PilzAdam 
boats, bones, doors

c55
creative, dye, wool

@webdesigner97 
farming

Ironzorg @VanessaE 
flowers

@PilzAdam @ShadowNinja 
tnt

xyz
xpanes

**Comment by rubenwardy** [comment]
Didn't @PilzAdam make farming?

**Comment by celeron55** [comment]
MIT.

**Comment by VanessaE** [comment]
Fine by me.

**Comment by paramat** [comment]
Aha https://github.com/minetest/minetest_game/commit/60ccb522d44c1a62b83fef886e1959d3898491fa
Se the changes to README, i think we should have both names in the copyright?

**Comment by xyzz** [comment]
Either is fine.

**Comment by sofar** [comment]
MIT is fine for doors and TNT from my part.

**Comment by paramat** [comment]
#1277 (part 1) merged. Part 2 coming.
Noticed default also needs work.

**Comment by paramat** [comment]
Original tnt node textures were by c55.

**Comment by paramat** [comment]
Tnt: All gunpowder textures except 'inventory' are by Wuzzy2.

**Comment by paramat** [comment]
Part 2 WIP #1278

**Comment by paramat** [comment]
I have some problems and questions, see #1278

**Comment by paramat** [comment]
#1278 merged but more to do and no doubt corrections to be done.

**Comment by paramat** [comment]
Let me check something, in default for example:
```
License of media (textures and sounds)
--------------------------------------
Attribution-ShareAlike 3.0 Unported (CC BY-SA 3.0)
http://creativecommons.org/licenses/by-sa/3.0/

Authors of media files
-----------------------
Everything not listed in here:
Copyright (C) 2010-2012 celeron55, Perttu Ahola <celeron55@gmail.com>

Cisoun's MIT texture pack:
  default_jungletree.png
  default_lava.png
  default_leaves.png
  default_sapling.png
  default_stone.png
  default_tree.png
  default_tree_top.png
  default_water.png

Cisoun's conifers mod (MIT):
  default_pine_needles.png
..
..
..
```
Does this mean that all media has been relicensed under CC BY-SA 3.0 and that the individual texture pack and sound licenses listed afterwards are the licenses before relicensing and therefore do not have to be included in the license.txt?
So in license.txt, for the media license i only include details of CC BY-SA 3.0 and do not need to include details of MIT etc.?

Or, perhaps this is just a mess, for example some textures have CC BY-SA 4.0 and therefore cannot be relicensed under CC BY-SA 3.0.

In my PR i have been adding license details for every license mentioned in README.txt, maybe some details are not required?

I'm coming across conflicting information like this a lot, it will state media as CC BY-SA 3.0, then under 'authors of media' a different license is mentioned. Which applies? Which needs to be detailed?

**Comment by rubenwardy** [comment]
You should use CC0 for textures and MIT for code

**Comment by sofar** [comment]
`CC-BY-3.0, CC-BY-4.0, CC-BY-SA-3.0, CC-BY-SA-4.0` are all perfectly OK for textures/artwork. I'd be careful not to inadvertently relicense textures from `CC-BY-*` to MIT.

I think you can relicense artwork from `MIT` to `CC-BY-3.0`, since as far as I can see that is allowed. The other way around may be problematic. You'd still want to retain the MIT header to be safe as well.

As for the general `License of media (textures and sounds)` statement, it is a *fallback* only, and IMHO the *preferred* license for artwork. Anything that specifically states a different license is whatever it states.

**Comment by paramat** [comment]
Okay thanks. We can't use CC0 for all textures because many are licensed under something else.

**Comment by rubenwardy** [comment]
You should use CC0 for textures *that was previously wtfpl as MIT is best for code*

**Comment by paramat** [comment]
Okay.

**Comment by sofar** [comment]
I think we should avoid CC0 entirely and relicense textures as CC-BY-3.0 (or even SA and/or 4.0) by default. CC0 is problematic enough that they should be avoided. The biggest problem is that if someone wrote `copyright John Smith` above `licensed under the WTFPL` it is conflicting, and we should avoid that conflict. We do that by erring on the safer side of the problem by *retaining* the copyright on the work, not by removing the copyright. CC0 effectively (implies) removal/relinquishing of the copyright, and so it may conflict with the *intent* of the original author. By relicensing it to CC-BY-*, we avoid that conflict, while still allowing the original author to *relicense it to CC0 themselves* irregardless.

TL:DR; don't relicense WTFPL to CC0, use CC-BY-* instead.

**Comment by paramat** [comment]
Good point, i'd prefer CC BY-SA 3.0 as it used as the standard and fallback, will do this.

**Comment by paramat** [comment]
#1281 merged.
Task mostly done but there may be some issues or corrections needed.
Will close this in a while, not now.

**Comment by paramat** [comment]
#1283 merged.
Cleanup and corrections.
Closing, will re-open if issues arise.
